File: Greedy/Chocola/Chocola.py
def min_cost_to_cut(cost_ver, cost_hor):
    # sort in descending order
    cost_ver.sort(reverse=True)
    cost_hor.sort(reverse=True)

    h = v = 0      
    hp = vp = 1     
    cost = 0

    while h < len(cost_hor) and v < len(cost_ver):
        if cost_hor[h] >= cost_ver[v]:
            cost += cost_hor[h] * vp
            h += 1
            hp += 1
        else:
            cost += cost_ver[v] * hp
            v += 1
            vp += 1

    while h < len(cost_hor):
        cost += cost_hor[h] * vp
        h += 1
        hp += 1

    while v < len(cost_ver):
        cost += cost_ver[v] * hp
        v += 1
        vp += 1

    return cost


# example 
cost_ver = [2, 1, 3, 1, 4]   # vertical cuts
cost_hor = [4, 1, 2]   # horizontal cuts

print("Total minimum cost:", min_cost_to_cut(cost_ver, cost_hor))

File: Greedy/Chocola/Chocola.js
function minCostToCut(costVer, costHor) {
    // sort in descending order
    costVer.sort((a, b) => b - a);
    costHor.sort((a, b) => b - a);

    let h = 0, v = 0, hp = 1, vp = 1;
    let cost = 0;

    while (h < costHor.length && v < costVer.length) {
        if (costHor[h] >= costVer[v]) {
            cost += costHor[h] * vp;
            h++; hp++;
        } else {
            cost += costVer[v] * hp;
            v++; vp++;
        }
    }

    while (h < costHor.length) cost += costHor[h++] * vp;
    while (v < costVer.length) cost += costVer[v++] * hp;

    return cost;
}

const costVer = [2, 1, 3, 1, 4];    // vertical cuts
const costHor = [4, 1, 2];    // horizontal cuts

console.log("Total minimum cost:", minCostToCut(costVer, costHor));
